### PR TITLE
rezabekf/cfn nag patch 3

### PR DESCRIPTION
**Pull request by rezabekf**
*Description of changes:*
This PR scopes down egress rules where possible. Please see in the comments the rules that weren't scoped down.

### CODE REVIEW

**Comment by rezabekf** [comment]
I wasn't able to figure out how to scope [down this egress rule](https://github.com/aws-samples/amazon-rds-proxy-multitenant-load-test/blob/03f8f8d695a91c9ac268c39fa9988470001a30d8/cfn/add-users.template#L141-L149)

 have tried:
- restricting to both SG group and port 3306 - fail
- restricting to SG and have all ports open - fail
- restricting to port 3306 and have CIdrIp open to world -fail

**Comment by rezabekf** [comment]
I have tried to restrict the [following egress rule](https://github.com/aws-samples/amazon-rds-proxy-multitenant-load-test/blob/03f8f8d695a91c9ac268c39fa9988470001a30d8/cfn/rds.template#L105-L113)

Two issues are happening here:
- it introduces circular dependency between `RDSSecurityGroup` and `ProxySecurityGroup` as the first one is `Ref` in the second one already.
- there is a problem when creating conditional statement for `RDSSecurityGroup`
```
RDSSecurityGroup:
    Type: AWS::EC2::SecurityGroup
    Properties:
      GroupDescription: RDS Security Group
      SecurityGroupEgress: !If
        - IsRDSProxy
        - - DestinationSecurityGroupId: !Ref ProxySecurityGroup
          - CidrIp: 0.0.0.0/0
            Description: Allow db outbound traffic
            IpProtocol: tcp
            FromPort: 3306
            ToPort: 3306
      VpcId: !Ref VpcID
```
The same code is used for both Proxy and No Proxy database and when NO proxy database is created the `cfn-lint` highlights the issue that resource `ProxySecurityGroup` may not be available when condition `IsRDSProxy` is False

I can mitigate this to add dedicated Security groups for Proxy and NoProxy db and have couple conditions to decide what gets created when
```
Conditions:
  IsRDSProxy: !Equals [!Ref CreateRDSProxy, true]
  IsNotRDSProxy: !Equals [!Ref CreateRDSProxy, false]
```
However, the NonProxy SG will still have egress open to the world, but I guess i can scope down port to 3306.

Port scoped down, see commit below.

File: cfn/ec2.template [hunk 1]
```diff
@@ -31,11 +31,23 @@ Resources:
...
+          ToPort: 80
+        - CidrIp: 0.0.0.0/0
+          Description: Allow all dns outbound traffic
+          IpProtocol: tcp
+          FromPort: 51
+          ToPort: 51
       SecurityGroupIngress:
         - CidrIp: !Ref OnPremIp
-          Description: allow locust port
+          Description: Allow http inboud traffic
```
Comment (HieronymusLex) [inline cfn/ec2.template]:
```suggestion
          Description: Allow http inbound traffic
```

File: cfn/ec2.template [hunk 2]
```diff
@@ -31,11 +31,23 @@ Resources:
...
+          ToPort: 443
+        - CidrIp: 0.0.0.0/0
+          Description: Allow all http outbound traffic
+          IpProtocol: tcp
+          FromPort: 80
+          ToPort: 80
+        - CidrIp: 0.0.0.0/0
+          Description: Allow all dns outbound traffic
+          IpProtocol: tcp
+          FromPort: 51
```
Comment (HieronymusLex) [inline cfn/ec2.template]: port 53 and UDP required

File: cfn/ec2.template [hunk 3]
```diff
@@ -31,11 +31,23 @@ Resources:
...
+          Description: Allow all https outbound traffic
+          IpProtocol: tcp
+          FromPort: 443
+          ToPort: 443
+        - CidrIp: 0.0.0.0/0
+          Description: Allow all http outbound traffic
+          IpProtocol: tcp
+          FromPort: 80
+          ToPort: 80
+        - CidrIp: 0.0.0.0/0
```
Comment (HieronymusLex) [inline cfn/ec2.template]: is 0.0.0.0/0 required here if not using custom dns resolvers?

**Review by HieronymusLex — CHANGES_REQUESTED** [review]

File: cfn/ec2.template (hunk 2 above)
Comment (rezabekf) [reply cfn/ec2.template]: Fixed port number and added UDP protocol

File: cfn/ec2.template (hunk 3 above)
Comment (rezabekf) [reply cfn/ec2.template]: Updated to Amazon provided DNS server.

**Review by HieronymusLex — APPROVED** [review]

**Pull request merged**